src/hw_traits/eusci.rs

```diff
@@ -1,237 +1,1083 @@
+                self.ucoe().bit()
+            }
+
+            #[inline(always)]
+            fn ucpe(&self) -> bool {
+                self.ucpe().bit()
+            }
+
+            #[inline(always)]
+            fn ucbrk(&self) -> bool {
+                self.ucbrk().bit()
+            }
+
+            #[inline(always)]
+            fn ucbusy(&self) -> bool {
+                self.ucbusy().bit()
+            }
+        }
+
+    };
+}
+
+macro_rules! eusci_b_impl {
+    ($intr_vec:ident, $EUsci:ident, $eusci:ident, $ucbxctlw0:ident, $ucbxctlw1:ident, $ucbxbrw:ident,
+     $ucbxstatw:ident, $ucbxtbcnt:ident, $ucbxrxbuf:ident, $ucbxtxbuf:ident, $ucbxi2coa0:ident,
+     $ucbxi2coa1:ident, $ucbxi2coa2:ident, $ucbxi2coa3:ident, $ucbxaddrx:ident, $ucbxaddmask:ident,
+     $ucbxi2csa:ident, $ucbxie:ident,
+     $ucbxifg:ident, $ucbxiv:ident, $Statw:ty, $Ifg:ty,
+     $StatwSpi:ty,
+     $ucbxctlw0spi:ident, $ucbxstatwspi:ident, $ucbxiespi:ident, $ucbxifgspi:ident) => {
+        eusci_impl!(
+            $intr_vec,
+            $EUsci,
+            $eusci,
+            $ucbxctlw0spi,
+            $ucbxctlw1,
+            $ucbxbrw,
+            $ucbxstatwspi,
+            $ucbxrxbuf,
+            $ucbxtxbuf,
+            $ucbxiespi,
+            $ucbxifgspi,
+            $ucbxiv,
+            $StatwSpi
+        );
+
+
+
+        impl EUsciI2C for pac::$EUsci {
+            type IfgOut = $Ifg;
+
+            #[inline(always)]
+            fn ctw0_rd_rst(&self) -> bool{
+                self.$ucbxctlw0().read().ucswrst().bit()
+            }
+
+            #[inline(always)]
+            fn ctw0_wr_rst(&self, bit:bool){
+                self.$ucbxctlw0().modify(|_, w| w.ucswrst().bit(bit))
+            }
+
+            #[inline(always)]
+            fn transmit_ack(&self){
+                self.$ucbxctlw0().modify(|_, w| w.uctxack().bit(true))
+            }
+
+            #[inline(always)]
+            fn transmit_nack(&self){
+                self.$ucbxctlw0().modify(|_, w| w.uctxnack().bit(true))
+            }
+
+            #[inline(always)]
+            fn transmit_start(&self){
+                self.$ucbxctlw0().modify(|_, w| w.uctxstt().bit(true))
+            }
+
+            #[inline(always)]
+            fn transmit_stop(&self){
+                self.$ucbxctlw0().modify(|_, w| w.uctxstp().bit(true))
+            }
+
+            #[inline(always)]
+            fn uctxstt_rd(&self) -> bool{
+                self.$ucbxctlw0().read().uctxstp().bit()
+            }
+
+            #[inline(always)]
+            fn uctxstp_rd(&self) -> bool{
+                self.$ucbxctlw0().read().uctxstp().bit()
+            }
+
+            #[inline(always)]
+            fn set_ucsla10(&self, bit:bool){
+                self.$ucbxctlw0().modify(|_, w| w.ucsla10().bit(bit))
+            }
+
+            #[inline(always)]
+            fn set_uctr(&self, bit:bool){
+                self.$ucbxctlw0().modify(|_, w| w.uctr().bit(bit))

```

Comment:
Replace all uses of `modify` with `set_bits`, `clear_bits`, or `toggle_bits`. These 3 APIs are atomic ways to modify the register, unlike `modify`, which does a read and a write and can run into race conditions.